src/AspNet.Security.OAuth.SuperOffice/SuperOfficeAuthenticationHandler.cs

```diff
@@ -0,0 +1,159 @@
+
+                throw new HttpRequestException($"An error occurred when retrieving SuperOffice user information ({response.StatusCode}). Please check if the authentication information is correct and the corresponding SuperOffice API is enabled.");
+            }
+
+            using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
+            var context = new OAuthCreatingTicketContext(new ClaimsPrincipal(identity), properties, Context, Scheme, Options, Backchannel, tokens, payload.RootElement);
+
+            if (Options.IncludeFunctionalRightsAsClaims
+                && !Options.ClaimActions.Any(c => c.ClaimType == SuperOfficeAuthenticationConstants.PrincipalNames.FunctionRights))
+            {
+                Options.ClaimActions.MapJsonKey(SuperOfficeAuthenticationConstants.PrincipalNames.FunctionRights, SuperOfficeAuthenticationConstants.PrincipalNames.FunctionRights);
+            }
+
+            context.RunClaimActions();
+            await Events.CreatingTicket(context);
+            return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
+        }
+
+        private async Task<ClaimsPrincipal> ValidateAsync(string idToken)
+        {
+            if (!_tokenHandler.CanValidateToken)
+            {
+                throw new NotSupportedException($"The configured {nameof(JwtSecurityTokenHandler)} cannot validate tokens.");
+            }
+
+            if (Options.ConfigurationManager == null)
+            {
+                throw new InvalidOperationException($"An OpenID Connect configuration manager has not been set on the {nameof(SuperOfficeAuthenticationOptions)} instance.");
+            }
+
+            var openIdConnectConfiguration = await Options.ConfigurationManager.GetConfigurationAsync(Context.RequestAborted);
+            Options.TokenValidationParameters.IssuerSigningKeys = openIdConnectConfiguration.JsonWebKeySet.Keys;
+
+            try
+            {
+                return _tokenHandler.ValidateToken(idToken, Options.TokenValidationParameters, out var _);
+            }
+            catch (Exception ex)
+            {
+                throw new SecurityTokenValidationException(
+                    "SuperOffice ID token validation failed for issuer {TokenIssuer} and audience {TokenAudience}.", ex);
+            }
+        }
+
+        /// <summary>
+        /// Store id_token in <paramref name="properties"/> token collection.
+        /// </summary>
+        /// <param name="properties">Authentication properties.</param>
+        /// <param name="idToken">The id_token JWT.</param>
+        private static void SaveIdToken(
+            [NotNull] AuthenticationProperties properties,
+            [NotNull] string idToken)
+        {
+            if (!string.IsNullOrWhiteSpace(idToken))
+            {
+                // save existing tokens, which are removed in StoreTokens method.
```

Comment:
```suggestion
                // Save existing tokens, which are removed in StoreTokens method.
```

Reply:
Must apologize, not used to this strict inner code comment styling. :-/